core-primitives/test/src/mock/ocall_api_mock.rs

```diff
@@ -0,0 +1,166 @@
+	EnclaveSidechainOCallApi,
+};
+use itp_teerex_storage::{TeeRexStorage, TeerexStorageKeys};
+use itp_types::{BlockHash, Enclave, ShardIdentifier, WorkerRequest, WorkerResponse};
+use sgx_types::{
+	sgx_epid_group_id_t, sgx_measurement_t, sgx_platform_info_t, sgx_quote_nonce_t,
+	sgx_quote_sign_type_t, sgx_report_t, sgx_spid_t, sgx_target_info_t, sgx_update_info_bit_t,
+	SgxResult, SGX_HASH_SIZE,
+};
+use sp_runtime::{AccountId32, OpaqueExtrinsic};
+use sp_std::prelude::*;
+use std::collections::HashMap;
+
+#[derive(Default, Clone, Debug)]
+pub struct OcallApiMock {
+	inner: HashMap<Vec<u8>, Vec<u8>>,
+	mr_enclave: [u8; SGX_HASH_SIZE],
+}
+
+impl OcallApiMock {
+	pub fn with_storage_entries<V: Encode>(mut self, entries: Vec<(Vec<u8>, V)>) -> Self {
+		for (k, v) in entries.into_iter() {
+			self.inner.insert(k, v.encode());
+		}
+		self
+	}
+
+	pub fn with_validateer_set(mut self, set: Option<Vec<Enclave>>) -> Self {
+		let set = set.unwrap_or_else(validateer_set);
+		self.inner.insert(TeeRexStorage::enclave_count(), (set.len() as u64).encode());
+		self.with_storage_entries(into_key_value_storage(set))
+	}
+
+	pub fn with_mr_enclave(mut self, mr_enclave: [u8; SGX_HASH_SIZE]) -> Self {
+		self.mr_enclave = mr_enclave;
+		self
+	}
+
+	pub fn insert(&mut self, key: Vec<u8>, value: Vec<u8>) {
+		self.inner.insert(key, value);
+	}
+
+	pub fn get(&self, key: &[u8]) -> Option<&Vec<u8>> {
+		self.inner.get(key)
+	}
+}
+
+impl EnclaveAttestationOCallApi for OcallApiMock {
+	fn sgx_init_quote(&self) -> SgxResult<(sgx_target_info_t, sgx_epid_group_id_t)> {
+		todo!()
+	}
+
+	fn get_ias_socket(&self) -> SgxResult<i32> {
+		Ok(42)
+	}
+
+	fn get_quote(
+		&self,
+		_sig_rl: Vec<u8>,
+		_report: sgx_report_t,
+		_sign_type: sgx_quote_sign_type_t,
+		_spid: sgx_spid_t,
+		_quote_nonce: sgx_quote_nonce_t,
+	) -> SgxResult<(sgx_report_t, Vec<u8>)> {
+		todo!()
+	}
+
+	fn get_update_info(
+		&self,
+		_platform_info: sgx_platform_info_t,
+		_enclave_trusted: i32,
+	) -> SgxResult<sgx_update_info_bit_t> {
+		todo!()
+	}
+
+	fn get_mrenclave_of_self(&self) -> SgxResult<sgx_measurement_t> {
+		Ok(sgx_measurement_t { m: self.mr_enclave })
+	}
+}
+
+impl EnclaveSidechainOCallApi for OcallApiMock {
+	fn propose_sidechain_blocks<SignedSidechainBlock: Encode>(
+		&self,
+		_signed_blocks: Vec<SignedSidechainBlock>,
+	) -> SgxResult<()> {
+		Ok(())
+	}
+
+	fn store_sidechain_blocks<SignedSidechainBlock: Encode>(
+		&self,
+		_signed_blocks: Vec<SignedSidechainBlock>,
+	) -> SgxResult<()> {
+		Ok(())
+	}
+
+	fn fetch_sidechain_blocks_from_peer<SignedSidechainBlock: Decode>(
+		&self,
+		_last_known_block_hash: BlockHash,
+		_shard_identifier: ShardIdentifier,
+	) -> SgxResult<Vec<SignedSidechainBlock>> {
+		Ok(Vec::new())
+	}
+}
+
+impl EnclaveMetricsOCallApi for OcallApiMock {
+	fn update_metric<Metric: Encode>(&self, _metric: Metric) -> SgxResult<()> {
+		Ok(())
+	}
+}
+
+// We cannot implement EnclaveOnChainOCallApi specifically here, because OnchainMock already
```

Comment:
Can you update the comments? Maybe also add a comment, why it was necessary to introduce a second ocall api mock.. respective what's the difference between them.